lib/at_cmd/at_cmd.c

```diff
@@ -76,6 +76,24 @@ static int open_socket(void)
 	return 0;
 }
 
+/*
+ * Do any validation of an AT command not performed by the lower layers or
+ * by the modem.
+ */
+static int check_cmd(const char *cmd) {
+	if (cmd == NULL) {
+		return -EINVAL;
+	}
+
+	// Check for presence of at least one printable non-whitespace character
+	for (char *c = cmd; c++; *c != '\0') {
```

Comment:
Shouldn't this be `for (char *c = cmd; *c != '\0'; c++) {`? `*c != '\0'` have no use here currently.

Reply:
Fixed. This would have been breaking, thanks for pointing it out.